packages/api/src/db/comment.ts

```diff
@@ -0,0 +1,69 @@
+import {RichTextNode} from '../graphql/richText'
+
+export enum CommentStatus {
+  Approved = 'approved',
+  PendingApproval = 'pendingApproval',
+  PendingUserChanges = 'pendingUserChanges',
+  Rejected = 'rejected'
+}
+
+export enum CommentRejectionReason {
+  Misconduct = 'misconduct',
+  Spam = 'spam'
+}
+
+export enum CommentAuthorType {
+  Admin = 'admin',
+  ArticleAuthor = 'articleAuthor',
+  VerifiedUser = 'verifiedUser',
+  PeerUser = 'peerUser',
+  Journalist = 'journalist',
+  Moderator = 'moderator'
+}
+
+export interface CommentData {
+  siteID?: string
+  userID: string
+  peerID?: string
+  permalink: string
+
+  // comment could be on different types of data models
+  articleID?: string
+  imageID?: string
+
+  revisions: [CommentRevision]
+  parentID?: string
+
+  authorType: CommentAuthorType
+  status: CommentStatus
+  rejectionReason: CommentRejectionReason
+}
+
+export interface Comment {
+  readonly id: string
+
+  readonly createdAt: Date
+  readonly modifiedAt: Date
+}
+
+export interface CommentRevision {
+  id: string
+  text: RichTextNode[]
+  createdAt: Date
+  modifiedAt: Date
```

Comment:
I noticed that here you define `modifiedAt` as mandatory but then you don't define it in `packages/api/src/graphql/comment.ts` without `GraphQLNonNull` (line 65). But maybe I misunderstood something?